packages/tools/src/utilities/math/polyline/getClosestLineSegmentIntersection.ts

```diff
@@ -0,0 +1,67 @@
+import { vec2 } from 'gl-matrix';
+import type { Types } from '@cornerstonejs/core';
+import lineSegmentsIntersect from './lineSegmentsIntersect';
+
+/**
+ * Checks whether the line (`p1`,`q1`) intersects any of the other lines in the
+ * `points`, and returns the closest value.
+ */
+export default function getClosestLineSegmentIntersection(
+  points: Types.Point2[],
+  p1: Types.Point2,
+  q1: Types.Point2,
+  closed = true
+): { segment: Types.Point2; distance: number } | undefined {
+  let initialI;
+  let j;
```

Comment:
let lastPointIndex instead makes this code more readable, similarly let currentPointIndex